README.md

```diff
@@ -1 +1,241 @@
+
+```shell
+$ pjw list
+1
+2
+3
+```
+
+get a specific job (and it's status).
+
+_Note: `end` and `exitCode` may be `null` if the job is still running_
+
+```shell
+$ pjw get 1
+{
+  "id": "1",
+  "command": "echo \"hello world\"",
+  "start": "1990-12-31T23:59:60Z",
+  "end": "1990-12-31T23:59:60Z",
+  "exitCode": 0,
+}
+```
+
+stop a job that's still running will return once the process has terminated
+
+```shell
+$ pjw stop 1
+{
+  "id": "1",
+  "command": "echo \"hello world\"",
+  "start": "1990-12-31T23:59:60Z",
+  "end": "1990-12-31T23:59:60Z",
+  "exitCode": 0,
+}
+```
+
+you can stream the job output. This always returns full results, starting from the beginning of the output, regardless of when you run the command
+
+```shell
+$ pjw logs 1
+hello world
+```
+
+### Mac M1 Host
+
+If you are running from a Mac M1/M2 that is using an ARM processor, you'll need virtualization that supports ARM such as [multipass](https://multipass.run/docs/mac-tutorial).
+
+```shell
+brew install multipass
+multipass launch 22.04
+```
+
+** _TODO More Steps Here_
+
+## Design
+
+### Scope
+
+#### In-Scope
+
+##### Library
+
+- Worker library with methods to start/stop/query status and get the output of a job.
+- Library should be able to stream the output of a running job.
+  - Output should be from start of process execution.
+  - Multiple concurrent clients should be supported.
+
+##### API
+
+- GRPC API to start/stop/get status/stream output of a running process.
+- Use mTLS authentication and verify client certificate. Set up strong set of cipher suites for TLS and good crypto setup for certificates. Do not use any other authentication protocols on top of mTLS.
+- Use a simple authorization scheme.
+- Add resource control for CPU, Memory and Disk IO per job using cgroups.
+- Add resource isolation for using PID, mount, and networking namespaces.
+
+##### CLI
+
+- CLI should be able to connect to worker service and start, stop, get status, and output of a job.
+
+#### Out-Of-Scope
+
+- HTTPS API to start/stop/get status of a running process.
+- Server/Client on separate machines
+- User-provided cgroup resources / isolation parameters
+- Persistence - Process logs are stored in memory, not persisted or read from disk
+- Scalability - Not designed to handle any sort of scale, just a simple demo
+- Availability - Not highly available
+- Cross-Platform - Only works on Linux AMD64
+- CI/CD pipeline
+
+### Server
+
+All job information is stored in memory. There's no persistence. If the server/application restarts, all job information and history is lost.
+
+A Job looks like this:
+
+```go
+type Job struct {
+  // for simplicity, Id is an int64, however, a GUID, UUID or other type might be better for a production-ready application
+  Id int64
+
+  // This is the `clientName` part of the client certificate CN
+  // Used to distinguish who created this job
+  ClientName string
+  Command string
```

Comment:
Is this a command line intended to be interpreted by a shell? "Command" is a bit ambiguous.